cmd/frontend/backend/repos.go

```diff
@@ -103,6 +119,42 @@ func (s *repos) GetByName(ctx context.Context, name api.RepoName) (_ *types.Repo
 	}
 }
 
+// checkAndUpdateRepo does a DB query to the gitserver_repos table for this repo to determine if there was an error
+// the last time this repo was updated. If there was no error, we pass through and do nothing, but if there was an
+// error we call the repo-updater service to update the repo. This code path handles the case where the
+// gitserverRepoStore was never initialized by just silently passing through.
+func (s *repos) checkAndUpdateRepo(ctx context.Context, repoID api.RepoID, name api.RepoName) {
+	if s.gitserverReposStore == nil {
+		// no gitserverRepoStore was initialized, so don't try to update the repo.
+		return
+	}
+	gsRepo, err := s.gitserverReposStore.GetByID(ctx, repoID)
+	if err != nil {
+		// Question: should we attempt to update the repo if we run into an error when fetching it from gitserver_repos?
+		log15.Error(fmt.Sprintf("Error getting repo from gitserver_repos: %s", err))
```

Comment:
```suggestion
		log15.Error("Error getting repo from gitserver_repos", "error", err))
```

We usually append a list of key value pairs when using log15